### PR TITLE
Display profit/loss for individual items

### LINKED ISSUE #4: Show profit/loss for every item

### PR DESCRIPTION
**Pull request by Toubsh**

### CODE REVIEW

File: src/app/features/crypto-item/crypto-item.component.ts [hunk 1]
```diff
@@ -8,20 +12,35 @@ import { CryptoItem } from './crypto-item.model';
...
   @Output() removeItem = new EventEmitter<CryptoItem>();
 
-  constructor() { }
+  constructor(private cryptoService: CryptoService) { }
 
   ngOnInit() {
     this.coin = this.item.coin;
     this.amount = this.item.amount;
     this.purchasePrice = this.item.purchasePrice;
+    this.subscription = interval(1000).subscribe(() => this.updatePrice());
```
Comment (Toubsh) [inline src/app/features/crypto-item/crypto-item.component.ts]: This still feels pretty wonky on initial load when creating a new check

**Comment by inpercima** [comment]
Wonderfull. Looks very good. :+1:  I will check this in next time.

File: src/app/features/crypto-item/crypto-item.component.html [hunk 2]
```diff
@@ -15,3 +15,7 @@
   </mat-form-field>
   <button fxFlex="20" fxFlexOffset.gt-xs="10" mat-raised-button color="primary" (click)="remove()">Remove</button>
 </div>
+<div class="profit-loss">
```
Comment (inpercima) [inline src/app/features/crypto-item/crypto-item.component.html]: Use mat-hint instead, see: https://material.angular.io/components/form-field/examples

File: src/app/features/crypto-item/crypto-item.component.css [hunk 3]
```diff
@@ -1,3 +1,7 @@
 .item {
   padding: 20px 0;
 }
+
+.profit-loss {
```
Comment (inpercima) [inline src/app/features/crypto-item/crypto-item.component.css]: Should be removed when mat-hint is used.

File: src/app/features/crypto-item/crypto-item.component.ts [hunk 4]
```diff
@@ -35,4 +54,8 @@ export class CryptoItemComponent implements OnInit {
     this.removeItem.emit(this.item);
   }
 
+  profitLoss(): number {
```
Comment (inpercima) [inline src/app/features/crypto-item/crypto-item.component.ts]: Could we reuse this function by moving to a service?

**Review by inpercima — CHANGES_REQUESTED** [review]

File: src/app/features/crypto-item/crypto-item.component.ts (hunk 4 above)
Comment (Toubsh) [reply src/app/features/crypto-item/crypto-item.component.ts]: Most probably

File: src/app/features/crypto-item/crypto-item.component.ts [hunk 5]
```diff
@@ -35,4 +53,12 @@ export class CryptoItemComponent implements OnInit {
...
+    return (this.cryptoPrices[this.item.coin + this.currency].value * this.item.amount) - (this.item.amount * this.item.purchasePrice);
+  }
+
+  conditionalColor() {
+    return this.profitLoss() < 0 ? { 'color': 'red' } : this.profitLoss() > 0 ? { 'color': 'green' } : { 'color': 'black' };
```
Comment (inpercima) [inline src/app/features/crypto-item/crypto-item.component.ts]: Unnecessarily quoted property 'color' found. (object-literal-key-quotes)tslint(1)

But regardless of this, we should outsource this to the css file and then specify the class accordingly.

**Review by inpercima — CHANGES_REQUESTED** [review]

**Review by inpercima — APPROVED** [review]

**Pull request merged**